Tests/TestingApps/PurchaseTesterSwiftUI/Shared/Helpers/PurchasesOrchestrator.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  PurchasesOrchestrator.swift
+//  PurchaseTester
+//
+//  Created by Nacho Soto on 12/19/22.
+//
+
+import RevenueCat
+import StoreKit
+
+/// Used to purchase products directly and test with observer mode.
+final class PurchasesOrchestrator: NSObject {
+
+    typealias PurchaseCompletedResult = SK1Transaction
+
+    private let paymentQueue: SKPaymentQueue
+
+    private var purchaseCompleteCallbacksByProductID: [String: (PurchaseCompletedResult) -> Void] = [:]
+
+    override init() {
+        self.paymentQueue = .init()
+
+        super.init()
+
+        self.paymentQueue.add(self)
+    }
+
+    deinit {
+        self.paymentQueue.remove(self)
+    }
+
+    func purchase(
+        sk1Product product: SK1Product,
+        completion: @escaping (PurchaseCompletedResult) -> Void
+    ) {
+        let productIdentifier = product.productIdentifier
+        assert(!productIdentifier.isEmpty)
+        assert(self.purchaseCompleteCallbacksByProductID[productIdentifier] == nil)
+
+        self.purchaseCompleteCallbacksByProductID[productIdentifier] = completion
+        self.paymentQueue.add(.init(product: product))
+    }
+
+    func purchase(sk2Product product: SK2Product) async throws {
+        let result = try await product.purchase()
+
+        switch result {
+        case let .success(.verified(transaction)):
+            await transaction.finish()
+
+            print("Successfully purchased SK2 product")
+        case let .success(.unverified(_, error)):
```

Comment:
do you not have to finish the transactions here?

Reply:
Hmm I guess we do yeah let me fix it.
It’s not a big deal cause this is just for internal testing, so I hope none of us is trying to cheat purchasing this 😅